resolvers/deno/sync.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+pub use inner::*;
+
+#[cfg(feature = "sync")]
+mod inner {
+  #![allow(clippy::disallowed_types)]
+
+  pub use std::sync::Arc as MaybeArc;
+
+  pub use dashmap::DashMap as MaybeDashMap;
+}
+
+#[cfg(not(feature = "sync"))]
+mod inner {
+  pub use std::rc::Rc as MaybeArc;
+
+  pub use std::collections::HashMap as MaybeDashMap;
```

Comment:
This is the right idea, but I think this will be a compile error because `HashMap` requires `&mut`? We might need to make a struct here with the same interface as DashMap.